monai/data/image_writer.py

```diff
@@ -106,17 +107,23 @@ def resolve_writer(ext_name, error_if_not_found=True) -> Sequence:
     if fmt.startswith("."):
         fmt = fmt[1:]
     avail_writers = []
+    required_pkg = []
     default_writers = SUPPORTED_WRITERS.get(EXT_WILDCARD, ())
     for _writer in look_up_option(fmt, SUPPORTED_WRITERS, default=default_writers):
         try:
             _writer()  # this triggers `monai.utils.module.require_pkg` to check the system availability
             avail_writers.append(_writer)
-        except OptionalImportError:
+        except OptionalImportError as e:
+            error_match = re.search(r"`(.*?)`", str(e))
+            if error_match:
+                required_pkg.append(error_match.group(1))
```

Comment:
I suggest you modify `OptionalImportError` to have a member recording the name of the package that isn't present so you don't have to do a re search. This would be used [here](https://github.com/Project-MONAI/MONAI/blob/e73257caa79309dcce1e93abf1632f4bfd75b11f/monai/utils/module.py#L473) like this `raise OptionalImportError(err_msg, pkg_name)` in place of the original line. The other place this is raised is [here](https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_writer.py#L119) where the second argument could just be `None`.

Reply:
@ericspod, If I understand you correctly:
1. The line [here](https://github.com/Project-MONAI/MONAI/blob/e73257caa79309dcce1e93abf1632f4bfd75b11f/monai/utils/module.py#L473) should be modified to
`raise OptionalImportError(err_msg, pkg_name)`
2. In image_writer I change this [line](https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_writer.py#L119) to
`raise OptionalImportError(f"No ImageWriter backend found for {fmt}.", None)`
3. Remove exception lines with re.search.
4. And to make it all works I should modify `OptionalImportError` module in monai/utils/module.py by adding a functionality to track package names that aren't present?

Reply:
> @ericspod, If I understand you correctly:
> 
>     1. The line [here](https://github.com/Project-MONAI/MONAI/blob/e73257caa79309dcce1e93abf1632f4bfd75b11f/monai/utils/module.py#L473) should be modified to
>        `raise OptionalImportError(err_msg, pkg_name)`
> 
>     2. In image_writer I change this [line](https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_writer.py#L119) to
>        `raise OptionalImportError(f"No ImageWriter backend found for {fmt}.", None)`
> 
>     3. Remove exception lines with re.search in image_writer.
> 
>     4. And to make it all works I should modify `OptionalImportError` module in monai/utils/module.py by adding a functionality to track package names that aren't present?

I think that's it, that'll allow you to get the name of the package without assumptions about the exception's error string, it'll be a more robust way of doing this going forward. Thanks!